shared/agent/src/providers/bitbucket.ts

```diff
@@ -1045,6 +1958,285 @@ export class BitbucketProvider
 		return response;
 	}
 
+	async removeReviewerFromPullRequest(request: {
+		reviewerId: string;
+		pullRequestId: string;
+		fullname: string;
+	}): Promise<Directives> {
+		const pr = await this.get<BitbucketPullRequest>(
+			`/repositories/${request.fullname}/pullrequests/${request.pullRequestId}`
+		);
+
+		let newReviewers: any[] = [];
+
+		if (pr.body.reviewers?.length === 1) {
+			newReviewers = [];
+		} else {
+			//remove that reviewer
+			pr.body.reviewers?.filter(_ => {
+				//@ts-ignore
+				if (_.account_id !== request.reviewerId) {
+					newReviewers.push(_);
+				}
+			});
+		}
+
+		const payload: { reviewers: any[] } = {
+			reviewers: newReviewers,
+		};
+		Logger.log(`commenting:removeRequestedReviewer`, {
+			request: request,
+			payload: payload,
+		});
+
+		const response = await this.put<BitbucketUpdateReviewerRequest, BitbucketPullRequest>(
+			`/repositories/${request.fullname}/pullrequests/${request.pullRequestId}`,
+			payload
+		);
+
+		const directives: Directive[] = [
+			{
+				type: "updatePullRequest",
+				data: {
+					updatedAt: Dates.toUtcIsoNow(),
+				},
+			},
+			{
+				type: "removeRequestedReviewer",
+				data: {
+					participants: response.body.participants,
+				},
+			},
+		];
+		return {
+			directives: directives,
+		};
+	}
+
+	async addReviewerToPullRequest(request: {
+		reviewerId: string;
+		pullRequestId: string;
+		fullname: string;
+	}): Promise<Directives> {
+		const pr = await this.get<BitbucketPullRequest>(
+			`/repositories/${request.fullname}/pullrequests/${request.pullRequestId}`
+		);
+
+		const repoSplit = request.fullname.split("/");
+		const workspace = repoSplit[0];
+
+		const members = await this.get<BitbucketValues<BitbucketWorkspaceMembers[]>>(
+			`/workspaces/${workspace}/members`
+		);
+
+		//get user info from members
+		let userInfo = pr.body.reviewers;
+		const selectedUser = request.reviewerId;
+		members.body.values.find(_ => {
+			if (_.user.account_id === selectedUser) {
+				//@ts-ignore
+				userInfo.push(_.user);
+			}
+		});
+
+		const newReviewers = userInfo;
+
+		const payload: any = {
+			reviewers: newReviewers,
+		};
```

Comment:
Since newReviewers is typed you don't need the any

```suggestion
		const payload = {
			reviewers: newReviewers ?? [],
		};```